isaac2/Assets/Scripts/Health.cs

```diff
@@ -0,0 +1,176 @@
+using System.Collections;
+using System.Collections.Generic;
+using Unity.VisualScripting;
+using UnityEngine;
+
+public class Health : MonoBehaviour
+{
+    /// <summary>
+    /// IDK C# DOCUMENTATION CONVENTIONS HELP!!!!!!!!!
+    /// 
+    /// Health class stores entity health and deals with setting, incrementing,
+    /// and fetching both current and max health.
+    /// Also detects death and overheal.
+    /// Used by other scripts.
+    /// </summary>
+
+    // TODO: if we want health bars they could be handled here
+
+    [SerializeField] private int maxHealth;  // or float?
+    [SerializeField] private int health;
+    
+    // Start is called before the first frame update
+    void Start()
+    {
+        health = maxHealth;
+        // Does maxhealth need to be initialized here ever? (Unity question)
+    }
+
+    // Update is called once per frame
+    //void Update()
+    //{
+        // Could implement gradual natural healing over time I guess?
+        // otherwise can we delete this method entirely (Unity question)
+    //}
+
+    /// <summary>
+    /// Returns current health of entity.
+    /// </summary>
+    /// <returns></returns>
+    public int GetHealth()
+    {
+        return health;
+    }
+
+    /// <summary>
+    /// Returns current maxHealth of entity.
+    /// </summary>
+    /// <returns></returns>
+    public int GetMaxHealth()
+    {
+        return maxHealth;
+    }
+
+    /// <summary>
+    /// Set current health to a new health value.
+    /// Prevents health modification to already dead entities.
+    /// Detects and prevents overheal.
+    /// Detects death and prevents negative health.
+    /// Returns new health in case caller needs (detect death etc.).
+    /// </summary>
+    /// <param name="amount"></param>
+    public int SetHealth(int newHealth) 
```

Comment:
I think this should be a private method too.

Reply:
Done